standard/classes.md

```diff
@@ -412,17 +412,18 @@ type_parameter_constraints
     ;
 
 primary_constraint
-    : class_type
-    | 'class'
+    : class_type '?'?
+    | 'class' '?'?
     | 'struct'
+    | 'notnull'
     | 'unmanaged'
     ;
 
 secondary_constraints
-    : interface_type
-    | type_parameter
-    | secondary_constraints ',' interface_type
-    | secondary_constraints ',' type_parameter
+    : interface_type '?'?
+    | type_parameter '?'?
```

Comment:
What is the intended meaning of *type_parameter* *nullable_type_attribute* as a constraint? The following text does not explain and while the current version of Roslyn accepts:

```
class Curious<S, T> where S : T?
```

It only accepts reference types for `S` & `T`.

Maybe some form of the (suggested removed) text on line 616 of types.md was meant to be here?

Reply:
Added a sentence to clarify.